From 04e349f679e98b2c0c6b57681a80a69027ad5294 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Wed, 16 May 2018 14:56:40 +0300
Subject: [PATCH 510/767] staging: fsl-dpaa2/eth: Restrict error message print

When trying to configure a distribution key, don't print
an error message if the operation is not supported (e.g.
when the interface is single-queue).

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index f3eee4c58fa8..eae730ee87f3 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -3015,7 +3015,8 @@ int dpaa2_eth_set_dist_key(struct dpaa2_eth_priv *priv,
 	dma_unmap_single(dev, key_iova, DPAA2_CLASSIFIER_DMA_SIZE,
 			 DMA_TO_DEVICE);
 	if (err) {
-		dev_err(dev, "Distribution key config failed\n");
+		if (err != -EOPNOTSUPP)
+			dev_err(dev, "Distribution key config failed\n");
 		goto free_key;
 	}
 
-- 
2.17.0

